File: Reverse_Words_in_a_String_III.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # Split the input string into words and reverse each word
        words = s.split()
        reversed_words = [word[::-1] for word in words]

        # Join the reversed words to form the final result
        return ' '.join(reversed_words)

File: Reverse_Words_in_a_String_III.cpp
class Solution {
public:
    string reverseWords(string s) {
        // Split the input string into words
        istringstream iss(s);
        vector<std::string> words;
        string word;
        while (iss >> word) {
            // Reverse each word and add to the vector
            reverse(word.begin(), word.end());
            words.push_back(word);
        }
        
        // Join the reversed words to form the final result
        string result = "";
        for (const string& reversedWord : words) {
            result += reversedWord + " ";
        }
        result.pop_back();  // Remove the extra space at the end
        
        return result;        
    }
};
